src/libpmem2/map_posix.c

```diff
@@ -352,9 +352,19 @@ pmem2_map(const struct pmem2_config *cfg, const struct pmem2_source *src,
 	 * MAP_FIXED - is required to mmap at exact address pointed by hint
 	 */
 	int flags = MAP_FIXED;
-	int proto = PROT_READ | PROT_WRITE;
 	void *addr;
 
+	/* "translate" pmem2 protection flags into linux flags */
+	int proto = 0;
+	if (PMEM2_PROT_EXEC & cfg->protection_flag)
+		proto = PROT_EXEC;
+	if (PMEM2_PROT_READ & cfg->protection_flag)
+		proto += PROT_READ;
+	if (PMEM2_PROT_WRITE & cfg->protection_flag)
+		proto += PROT_WRITE;
+	if (PMEM2_PROT_FROM_FD & cfg->protection_flag)
+		proto = 0;
+
 	if (file_type == PMEM2_FTYPE_DIR) {
```

Comment:
GitHub doesn't allow comments on lines outside of the diff, so I have to comment here:

The code around file_map also needs to be updated.